### PR TITLE
Use bs.obj

### PR DESCRIPTION
**Pull request by Gregoirevda**

### CODE REVIEW

**Comment by janicduplessis** [comment]
Nice! :)

File: src/ApolloInMemoryCache.re [hunk 1]
```diff
@@ -36,22 +36,16 @@ let createIntrospectionFragmentMatcher = (~data) =>
...
-    })
-    | (None, Some(fragmentMatcher)) => 
-    apolloInMemoryCache({
-      "fragmentMatcher": Js.Nullable.return(fragmentMatcher)
-    })
-    | (None, None) => apolloInMemoryCache(Js.Obj.empty())
-  };
+    /* Apollo Client, looks for key in Object. Doesn't check if value is null  */
+  apolloInMemoryCache(
+     makeApolloInMemoryCacheParams(~dataIdFromObject=?dataIdFromObject, ~fragmentMatcher=?fragmentMatcher)
```
Comment (thangngoc89) [inline src/ApolloInMemoryCache.re]: Short version: 

```
makeApolloInMemoryCacheParams(~dataIdFromObject?, ~fragmentMatcher?)
```

Comment (Gregoirevda) [reply src/ApolloInMemoryCache.re]: Ah that syntax! Thanks, I'll upgrade

File: src/ApolloInMemoryCache.re [hunk 2]
```diff
@@ -36,22 +36,16 @@ let createIntrospectionFragmentMatcher = (~data) =>
...
-    })
-    | (None, Some(fragmentMatcher)) => 
-    apolloInMemoryCache({
-      "fragmentMatcher": Js.Nullable.return(fragmentMatcher)
-    })
-    | (None, None) => apolloInMemoryCache(Js.Obj.empty())
-  };
+    /* Apollo Client, looks for key in Object. Doesn't check if value is null  */
+  apolloInMemoryCache(
+     makeApolloInMemoryCacheParams(~dataIdFromObject=?, ~fragmentMatcher=?)
```
Comment (thangngoc89) [inline src/ApolloInMemoryCache.re]: I’m not sure this is right. Punning optional args should be in this form 


makeApolloInMemoryCacheParams(~dataIdFromObject?, ~fragmentMatcher?)

Comment (thangngoc89) [reply src/ApolloInMemoryCache.re]: Notice no =

Comment (Gregoirevda) [reply src/ApolloInMemoryCache.re]: Thanks for pointing out!! https://github.com/apollographql/reason-apollo/commit/6cc713fb4f2601ee67976c94f0f91bd5b4ad33bc

**Pull request merged**